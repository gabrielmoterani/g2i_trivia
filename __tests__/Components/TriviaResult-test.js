import React from 'react';
import { shallow } from 'enzyme';
import TrivaResult from '../../src/components/TriviaResult';



initialProps = {
    questions:     
    [{
        "category": "Entertainment: Video Games",
        "type": "boolean",
        "difficulty": "hard",
        "question": "In Undertale, having a &quot;Fun Value&quot; set to 56-57 will play the &quot;Wrong Number Song Call&quot;.",
        "correct_answer": false,
        "incorrect_answers": [
          "True"
        ],
        "answerWasCorrect": true
    },
    {
        "category": "Science: Mathematics",
        "type": "boolean",
        "difficulty": "hard",
        "question": "The binary number &quot;101001101&quot; is equivalent to the Decimal number &quot;334&quot;",
        "correct_answer": false,
        "incorrect_answers": [
          "True"
        ],
        "answerWasCorrect": false
    }
    ]
}


const setup = (props = initialProps) => shallow(<TrivaResult  {...props} />);


describe('Test Result Card Component', () => {
    const wrapper = setup();
  
    it('should render properly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    
    it('should have questions', () => {
        const questions = wrapper.find('Text')
        expect(questions.length).toBeGreaterThanOrEqual(2);


    });

});