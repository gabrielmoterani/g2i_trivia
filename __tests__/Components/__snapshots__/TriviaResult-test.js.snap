// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Result Card Component should render properly 1`] = `
<ScrollViewMock
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 20,
      "marginVertical": 50,
    }
  }
>
  <View
    key="0"
  >
    <View
      style={
        Object {
          "color": "#49dab4",
          "fontSize": 15,
          "fontWeight": "500",
          "marginBottom": 10,
          "textAlign": "justify",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#49dab4",
            "fontSize": 15,
            "fontWeight": "500",
            "marginBottom": 10,
            "textAlign": "justify",
          }
        }
      >
        +
         
         
        In Undertale, having a "Fun Value" set to 56-57 will play the "Wrong Number Song Call".
      </Text>
    </View>
  </View>
  <View
    key="1"
  >
    <View
      style={
        Object {
          "color": "#ff3859",
          "fontSize": 15,
          "fontWeight": "500",
          "marginBottom": 10,
          "textAlign": "justify",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ff3859",
            "fontSize": 15,
            "fontWeight": "500",
            "marginBottom": 10,
            "textAlign": "justify",
          }
        }
      >
        -
         
         
        The binary number "101001101" is equivalent to the Decimal number "334"
      </Text>
    </View>
  </View>
</ScrollViewMock>
`;
